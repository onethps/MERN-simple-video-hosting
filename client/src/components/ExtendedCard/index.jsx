import React from 'react';
import {
  Base,
  Container,
  Desc,
  ECardLink,
  ProfileBlock,
  ProfileImage,
  ProfileName,
  Thumbnail,
  Title,
  ViewsAndData,
} from './styles/extendedcard';

export default function ExtendedCard({ children }) {
  return <Container>{children}</Container>;
}

ExtendedCard.Thumbnail = function ExtendedCardThumbnail({ src, ...restProps }) {
  return <Thumbnail {...restProps} src={src} />;
};

ExtendedCard.Base = function ExtendedCardBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

ExtendedCard.Title = function ExtendedCardTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

ExtendedCard.Link = function ExtendedCardLink({ children, to, ...restProps }) {
  return (
    <ECardLink to={to} {...restProps}>
      {children}
    </ECardLink>
  );
};

ExtendedCard.ProfileFrame = function ExtendedCardProfileFrame({
  children,
  ...restProps
}) {
  return <ProfileBlock {...restProps}>{children}</ProfileBlock>;
};

ExtendedCard.ProfileImage = function ExtendedCardProfileImage({ src, ...restProps }) {
  return <ProfileImage src={src} {...restProps} />;
};

ExtendedCard.ProfileName = function ExtendedCardProfileName({ children, ...restProps }) {
  return <ProfileName {...restProps}>{children}</ProfileName>;
};

ExtendedCard.ViewsAndData = function ExtendedCardViewsAndData({
  children,
  ...restProps
}) {
  return <ViewsAndData {...restProps}>{children}</ViewsAndData>;
};

ExtendedCard.Description = function ExtendedCardDescription({ children, ...restProps }) {
  return <Desc {...restProps}>{children}</Desc>;
};
