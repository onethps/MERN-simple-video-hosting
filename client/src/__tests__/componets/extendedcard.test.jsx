import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import ExtendedCard from 'components/ExtendedCard';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { format } from 'timeago.js';

describe('<Card />', () => {
  const type = 'sm';

  const user = {
    _id: '55',
    name: 'Victor',
  };

  const desc =
    'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur dolores eius expedita harum illum impedit iatem';

  const desc1 =
    'Lorem ipsumww dolor sit amet, consectetur adipisicing elit. Aspernatur dolores eius expedita harum illum impedit iatem';

  const cardVideos = [
    {
      _id: '1',
      imgUrl: 'https://www.google.com/1',
      title: 'Video 1',
      views: '10',
      userId: '55',
      desc: desc,
      createdAt: new Date().toLocaleString(),
    },
    {
      _id: '2',
      imgUrl: 'https://www.google.com/2',
      title: 'Video 2',
      views: '50',
      userId: '12',
      desc: desc1,
      createdAt: new Date().toLocaleString(),
    },
  ];

  it('renders the <ExtendedCard /> with populated data', () => {
    jest.mock('react-router-dom', () => {
      return {
        Redirect: jest.fn(({ to }) => `Redirected to ${to}`),
      };
    });

    const { getByText, container } = render(
      <Router>
        <>
          {cardVideos.map((video) => {
            return (
              <ExtendedCard key={video._id}>
                <ExtendedCard.Link
                  to={`/video/${video?._id}`}
                  data-testid="link-test-thumbnail"
                >
                  <ExtendedCard.Thumbnail src={video?.imgUrl} />
                </ExtendedCard.Link>
                <ExtendedCard.Base>
                  <ExtendedCard.Title>{video?.title}</ExtendedCard.Title>
                  <ExtendedCard.ProfileFrame>
                    <ExtendedCard.ProfileImage src={user?.img} />
                    <ExtendedCard.ProfileName>{user?.name}</ExtendedCard.ProfileName>
                  </ExtendedCard.ProfileFrame>
                  <ExtendedCard.ViewsAndData>
                    {video?.views} views - {format(video?.createdAt)}
                  </ExtendedCard.ViewsAndData>
                  <ExtendedCard.Description>{video?.desc}</ExtendedCard.Description>
                </ExtendedCard.Base>
              </ExtendedCard>
            );
          })}
        </>
      </Router>,
    );

    expect(getByText('Video 1')).toBeInTheDocument();
    expect(getByText('50 views - just now')).toBeInTheDocument();
    expect(getByText('Video 2')).toBeInTheDocument();
    expect(getByText('10 views - just now')).toBeInTheDocument();
    expect(getByText('10 views - just now')).toBeInTheDocument();
    expect(getByText(desc)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
