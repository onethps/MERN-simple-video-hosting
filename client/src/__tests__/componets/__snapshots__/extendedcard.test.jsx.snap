// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <ExtendedCard /> with populated data 1`] = `
<div
  class="sc-bczRLJ eqVzob"
>
  <a
    class="sc-hKMtZM eAFTQK"
    data-testid="link-test-thumbnail"
    href="/video/1"
  >
    <img
      class="sc-gsnTZi iNecrN"
      src="https://www.google.com/1"
    />
  </a>
  <div
    class="sc-dkzDqf knZSwS"
  >
    <h1
      class="sc-eCYdqJ TnSYU"
    >
      Video 1
    </h1>
    <div
      class="sc-gKXOVf cvKPfw"
    >
      <img
        class="sc-iBkjds ixWmXM"
      />
      <span
        class="sc-ftvSup dTLfxu"
      >
        Victor
      </span>
    </div>
    <span
      class="sc-jSMfEi iOHjy"
    >
      10
       views - 
      just now
    </span>
    <p
      class="sc-papXJ jtkNPH"
    >
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aspernatur dolores eius expedita harum illum impedit iatem
    </p>
  </div>
</div>
`;
